src/proptest.rs

```diff
@@ -0,0 +1,95 @@
+use std::iter::zip;
+
+use crate::{Direction, PortGraph, PortIndex};
+use proptest::prelude::*;
+use rand::seq::SliceRandom;
+
+prop_compose! {
+    /// A random graph with random number of ports but no edges
+    ///
+    /// The graph has at least one node.
+    ///
+    ///  - `max_n_nodes` is the maximum number of nodes in the generated graph
+    ///  - `max_port` is the maximum number of incoming and outgoing ports at
+    ///    every node
+    fn gen_no_edge_graph(max_n_nodes: usize, max_port: usize)(
+        ports in prop::collection::vec(0..=max_port, 2..=max_n_nodes)
+    ) -> PortGraph {
+        let mut g = PortGraph::new();
+        let mut ind = 0;
+        while ind + 1 < ports.len() {
+            g.add_node(ports[ind], ports[ind + 1]);
+            ind += 2;
```

Comment:
Doesn't this generate `max_n_nodes/2` nodes at max? Perhaps generating `(# inputs, # outputs)` tuples for `ports`, and having a `for` loop here would be clearer.

Reply:
Yep, you're right. Fixed.